### PR TITLE
support https (#175)

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Solve Issue #175 and return make get_flask_app public

### CODE REVIEW

File: taipy/gui/gui.py [hunk 1]
```diff
@@ -1109,15 +1121,17 @@ def run(self, run_server: bool = True, run_in_thread: bool = False, **kwargs) ->
         self._bindings()._set_single_client(bool(app_config["single_client"]))
 
         # Start Flask Server
```
Comment (FredLL-Avaiga) [inline taipy/gui/gui.py:1134]: Need to update the return type of the run method and the doc ?

**Review by FredLL-Avaiga — APPROVED** [review]
need a bit of doc :-)

File: taipy/gui/gui.py (hunk 1 above)
Comment (dinhlongviolin1) [reply taipy/gui/gui.py:1134]: Thanks! I just update the types before seeing this message :D Will update the docs

**Review by FabienLelaquais — APPROVED** [review]
Looks just fine!

File: taipy/gui/gui.py [hunk 2]
```diff
@@ -960,15 +966,15 @@ def _navigate(self, to: t.Optional[str] = ""):
...
+    def get_flask_app(self):
         return self._server.get_flask()
 
     def _set_frame(self, frame: FrameType):
         if not isinstance(frame, FrameType):
             raise RuntimeError("frame must be a FrameType where Gui can collect the local variables.")
         self.__frame = frame
 
-    def run(self, run_server: bool = True, run_in_thread: bool = False, **kwargs) -> None:
+    def run(self, run_server: bool = True, run_in_thread: bool = False, **kwargs) -> t.Optional[Flask]:
```
Comment (florian-vuillemot) [inline taipy/gui/gui.py:986]: Why Optional ?

**Review by florian-vuillemot — APPROVED** [review]

File: taipy/gui/gui.py (hunk 2 above)
Comment (dinhlongviolin1) [reply taipy/gui/gui.py:986]: if run_server is True, it will return None

**Pull request merged**